### PR TITLE
Docs: Replace all absolute production links

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
The website contains multiple links to the production site which the docusaurus broken link checker won't pick up as it doesn't check absolute links to other websites.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

I've put this branch on our dev bucket and manually checked all the changed links in this PR. ✅

### CODE REVIEW

**Review by academo — APPROVED** [review]
Any chance we can add a linter rule to not allow absolute links to https://grafana.com/developers/plugin-tools/* ?

**Review by sympatheticmoose — COMMENTED** [review]
One wrong link to be addressed, otherwise LGTM 🎉

File: docusaurus/docs/how-to-guides/data-source-plugins/fetch-data-from-frontend.md [hunk 1]
```diff
@@ -249,7 +249,7 @@ const response = getBackendSrv().fetch<TODO[]>({
 
 ## Add authentication to your requests using the data proxy
 
-To learn about adding authentication to the data proxy, refer to our [documentation](https://grafana.com/developers/plugin-tools/how-to-guides/data-source-plugins/add-annotations-for-data-source-plugins) (./add-annotations-for-data-source-plugins).
+To learn about adding authentication to the data proxy, refer to our [documentation](./add-annotations-for-data-source-plugins.md).
```
[the remark this answers is not in the document]
Comment (jackw) [reply docusaurus/docs/how-to-guides/data-source-plugins/fetch-data-from-frontend.md]: Thanks for catching this @sympatheticmoose  🦅 👀

**Review by sympatheticmoose — APPROVED** [review]

File: docusaurus/docs/how-to-guides/data-source-plugins/convert-a-frontend-datasource-to-backend.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ keywords:
 
 This guide shows you how to convert an existing frontend-only data source plugin into a [backend plugin](../../key-concepts/backend-plugins).
 
-To convert the frontend data source, we recommend scaffolding a new backend data source plugin using [`npx @grafana/create-plugin@latest`](https://grafana.com/developers/plugin-tools/). Use the following instructions to extend this foundation to copy functionality from your original plugin.
+To convert the frontend data source, we recommend scaffolding a new backend data source plugin using [`npx @grafana/create-plugin@latest`](/). Use the following instructions to extend this foundation to copy functionality from your original plugin.
```
[the remark this answers is not in the document]
Comment (jackw) [reply docusaurus/docs/how-to-guides/data-source-plugins/convert-a-frontend-datasource-to-backend.md:16]: I have no idea. I've left it in as I'm just trying to fix things up. Should I remove before merge?

**Comment by jackw** [comment]
> Any chance we can add a linter rule to not allow absolute links to https://grafana.com/developers/plugin-tools/* ?

I'm looking into this as without something to block we're gonna end up in this state again at some point. 👍

**Pull request merged**